libraries/node-core-library/src/FileSystem.ts

```diff
@@ -785,6 +791,59 @@ export class FileSystem {
     });
   }
 
+  /**
+   * Writes the contents of multiple Uint8Arrays to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writevSync()`.
+   * This API is useful for writing large files efficiently, especially if the input is being concatenated from
+   * multiple sources.
+   * @remarks
+   * Throws an error if the folder doesn't exist, unless ensureFolder=true.
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The content that should be written to the file.
+   * @param options - Optional settings that can change the behavior.
+   */
+  public static writeBuffersToFile(
+    filePath: string,
+    contents: Iterable<Uint8Array>,
+    options?: IFileSystemWriteBinaryFileOptions
+  ): void {
+    FileSystem._wrapException(() => {
+      const toCopy: Uint8Array[] = [...contents];
+
+      let fd: number | undefined;
+      try {
+        fd = fsx.openSync(filePath, 'w');
+      } catch (error) {
+        if (!options?.ensureFolderExists || !FileSystem.isNotExistError(error as Error)) {
+          throw error;
```

Comment:
Without much extra code, we could avoid an extra `throw` here in the case where `ensureFolderExists === false`.

Reply:
I don't understand this comment. I'm in a `catch` block, therefore the error is currently suppressed, so to unsuppress it I need a `throw`.

Reply:
Something like:

```ts
if (options?.ensureFolderExists) {
  try {
    fd = fsx.openSync(filePath, 'w');
  } catch (error) {
    if (!FileSystem.isNotExistError(error as Error)) {
      throw error;
    }

    const folderPath: string = nodeJsPath.dirname(filePath);
    FileSystem.ensureFolder(folderPath);
    fd = fsx.openSync(filePath, 'w');
  }
} else {
  fd = fsx.openSync(filePath, 'w'); // no wasteful re-throw
}
```

Reply:
Keeping the current mode for readability. Optimizing for performance of the exception case seems wasteful.